### PR TITLE
Updates the CircleCI configuration to test against Ruby releases 2.6.3, 2.5.5, and 2.4.6 along with JRuby release 9.2.0.0

### LINKED ISSUE #120: Test against Ruby releases 2.6.3, 2.5.5, and 2.4.6 along with JRuby release 9.2.0.0
Derived from samvera/maintenance#17

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #120

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -65,15 +73,100 @@ workflows:
   ci:
     jobs:
       - bundle:
+          name: "bundle_ruby2-6_rails5-2"
```
Comment (jrgriffiniii) [inline .circleci/config.yml]: These jobs should be vertically integrated (please see https://github.com/samvera/om/pull/78#pullrequestreview-238616619)

**Comment by jrgriffiniii** [comment]
At this point the failing jobs in https://circleci.com/workflow-run/27a914c6-9227-4cab-98f1-2a715ff5141b are all due to an issue with jetty_wrapper triggering 503 responses from Fedora.

**Comment by jrgriffiniii** [comment]
`jetty_wrapper` seems to be failing in these jobs due to certain Java dependencies being unavailable:
```
INFO 2019-07-30 19:49:58.858 [main] (BasicServer) Fedora Version: 3.8.1
INFO 2019-07-30 19:49:58.858 [main] (BasicServer) Fedora Build Date: 2015-07-20 12:14:48 UTC-0400
INFO 2019-07-30 19:49:58.859 [main] (Server) Loading fcfg datastore definitions for localMySQLPool
INFO 2019-07-30 19:49:58.859 [main] (Server) Loading fcfg datastore definitions for localDerbyPool
INFO 2019-07-30 19:49:58.859 [main] (Server) Loading fcfg datastore definitions for localOraclePool
INFO 2019-07-30 19:49:58.860 [main] (Server) Loading fcfg datastore definitions for localPostgreSQLPool
INFO 2019-07-30 19:49:58.860 [main] (Server) Loading fcfg datastore definitions for localMulgaraTriplestore
INFO 2019-07-30 19:49:58.860 [main] (Server) Loading fcfg datastore definitions for localPostgresMPTTriplestore
INFO 2019-07-30 19:49:58.860 [main] (Server) Loading fcfg datastore definitions for apimUpdateMessages
INFO 2019-07-30 19:49:58.860 [main] (Server) Loading fcfg datastore definitions for apimAccessMessages
INFO 2019-07-30 19:49:58.860 [main] (Server) Loading bean definitions for org.fcrepo.server.security.DefaultAuthorization impl class=org.fcrepo.server.security.Authorization
INFO 2019-07-30 19:49:58.862 [main] (Server) Loading bean definitions for org.fcrepo.server.security.DefaultBackendSecurity impl class=org.fcrepo.server.security.BackendSecurity
INFO 2019-07-30 19:49:58.863 [main] (Server) Loading bean definitions for org.fcrepo.server.storage.DefaultDOManager impl class=org.fcrepo.server.storage.DOManager
INFO 2019-07-30 19:49:58.864 [main] (Server) Loading bean definitions for org.fcrepo.server.access.DefaultAccess impl class=org.fcrepo.server.access.Access
INFO 2019-07-30 19:49:58.866 [main] (Server) Loading bean definitions for org.fcrepo.server.access.DynamicAccessModule impl class=org.fcrepo.server.access.DynamicAccess
INFO 2019-07-30 19:49:58.866 [main] (Server) Loading bean definitions for org.fcrepo.server.search.FieldSearchSQLModule impl class=org.fcrepo.server.search.FieldSearch
INFO 2019-07-30 19:49:58.866 [main] (Server) Loading bean definitions for org.fcrepo.server.resourceIndex.ResourceIndexModule impl class=org.fcrepo.server.resourceIndex.ResourceIndex
INFO 2019-07-30 19:49:58.867 [main] (Server) Loading bean definitions for org.fcrepo.server.oai.FedoraOAIProviderModule impl class=org.fcrepo.oai.OAIProvider
INFO 2019-07-30 19:49:58.868 [main] (Server) Loading bean definitions for org.fcrepo.server.storage.translation.DOTranslatorModule impl class=org.fcrepo.server.storage.translation.DOTranslator
INFO 2019-07-30 19:49:58.868 [main] (Server) Loading bean definitions for org.fcrepo.server.management.BasicPIDGenerator impl class=org.fcrepo.server.management.PIDGenerator
INFO 2019-07-30 19:49:58.868 [main] (Server) Loading bean definitions for org.fcrepo.server.messaging.MessagingModule impl class=org.fcrepo.server.messaging.Messaging
INFO 2019-07-30 19:49:58.869 [main] (Server) Loading bean definitions for org.fcrepo.server.storage.ConnectionPoolManagerImpl impl class=org.fcrepo.server.storage.ConnectionPoolManager
INFO 2019-07-30 19:49:58.870 [main] (Server) Loading bean definitions for org.fcrepo.server.validation.DOValidatorModule impl class=org.fcrepo.server.validation.DOValidator
INFO 2019-07-30 19:49:58.870 [main] (Server) Loading bean definitions for org.fcrepo.server.storage.DefaultExternalContentManager impl class=org.fcrepo.server.storage.ExternalContentManager
INFO 2019-07-30 19:49:58.871 [main] (Server) Loading bean definitions for org.fcrepo.server.management.ManagementModule impl class=org.fcrepo.server.management.Management
ERROR 2019-07-30 19:49:58.919 [main] (Server) Fatal error while starting server
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.context.annotation.CommonAnnotationBeanPostProcessor': Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Could not instantiate bean class [org.springframework.context.annotation.CommonAnnotationBeanPostProcessor]: Constructor threw exception; nested exception is java.lang.NoClassDefFoundError: javax/annotation/PostConstruct
```

**Comment by jrgriffiniii** [comment]
Java EE 11 onwards might have removed `javax/annotation` with https://bugs.openjdk.java.net/browse/JDK-8189188

**Comment by jrgriffiniii** [comment]
https://github.com/samvera/rubydora/pull/121/commits/50515c84e3203ece62b5692dfa343827350b9504 finally gets this passing.  However, it does shift the dependency away from https://github.com/samvera-deprecated/hydra-jetty (which cannot be maintained) and towards a Docker image provided by https://hub.docker.com/r/docuteam/fedora3-gsearch.  Perhaps before proceeding docuteam should be contacted with a request to push the image to the `samvera` Docker Hub organization at https://hub.docker.com/u/samvera

**Comment by jrgriffiniii** [comment]
Reviewing this now, I'm not certain why I was testing against release 9.2.0.0 of JRuby.  I'm only seeing 9.2.7.0 and 9.1.17.0 specified as the latest releases on https://www.jruby.org/download.

Would @jcoyne or others object if I dropped support for 9.2.0.0 on this?

**Comment by jrgriffiniii** [comment]
This also needs to be rebased with the excellent new Dockerfile provided by @botimer in https://github.com/samvera/rubydora/pull/124

**Comment by jrgriffiniii** [comment]
https://github.com/samvera/rubydora/pull/121/commits/aad3d96953030482563c3cbef730e27b3424c1a7 failed on CircleCI.

The explanation for this, I believe, is that CircleCI runs containers for the test environments, and virtualizing Docker containers within a Docker environment is not permitted.  One alternative might be to use the `machine` executor (please see https://circleci.com/docs/2.0/executor-types/#using-machine), but this would require much more labor (and would possibly be temporary).

I would please propose that we look to contribute @botimer's work to a samvera-labs repository and publish the Docker images to https://hub.docker.com/u/samveralabs.  I can do this myself, but given that it is entirely botimer's work, I want them to have the opportunity to commit it under their account.

**Comment by jrgriffiniii** [comment]
> I would please propose that we look to contribute @botimer's work to a samvera-labs repository and publish the Docker images to https://hub.docker.com/u/samveralabs. I can do this myself, but given that it is entirely botimer's work, I want them to have the opportunity to commit it under their account.

I should also add that, if this is acceptable, I can volunteer to serve as the Product Owner.

**Comment by botimer** [comment]
@jrgriffiniii I am quite happy for you to take the image publication forward. The Dockerfile itself is contributed under my name and CLA. That is more than enough for me, assuming I will be asked about any maintenance! 😄

**Pull request merged**